src/libraries/System.Private.CoreLib/src/System/IO/BufferedStream.cs

```diff
@@ -209,14 +258,27 @@ public override long Position
                 EnsureCanSeek();
 
                 if (_writePos > 0)
-                    FlushWrite();
+                    FlushWrite(true);
 
                 _readPos = 0;
                 _readLen = 0;
                 _stream!.Seek(value, SeekOrigin.Begin);
             }
         }
 
+        // this method exists to keep old FileStream behaviour and don't perform a Flush when getting Position
+        internal long GetPositionWithoutFlushing()
+        {
+            Debug.Assert(_actLikeFileStream);
+
+            EnsureNotClosed();
+            EnsureCanSeek();
+
+            return (_stream!.Position - _readLen) + _readPos + _writePos;
+        }
```

Comment:
I don't see how the code here is any different from get_Position (other than an unrelated reorder of the arithmetic, and different debug-only asserts).  What am I missing?

Reply:
It looks like you are right and it was my mistake. Let me change it and see if anything gets broken ;)